File: BOJ/silver2/2023-02-12/10799.py
stk = []
ans = 0
st = input()
for i, ch in enumerate(st):
    if ch == '(' and st[i + 1] == ')':
        ans += len(stk)
    elif ch == ')' and st[i - 1] == '(':
        continue
    elif ch == '(':
        stk.append(ch)
    else:
        stk.pop()
        ans += 1
print(ans)

File: cpp/바킹독_알고리즘/스택의 활용(수식의 괄호 쌍)/10799.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    string s;
    cin >> s;
    stack<char> stk;
    long long cnt = 0;
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == '(') stk.push(s[i]);
        else {
            stk.pop();
            if (i - 1 >= 0 && s[i - 1] == '(') cnt += stk.size();
            else cnt++;
        }
    }
    cout << cnt;
}